### DISCUSSION TITLE
Failed to set 32 bits display mode - Linux Framebuffer embedded device

### DISCUSSION THREAD

**Discussion by TheDutchDev** [unoplatform/uno#8737] [Q&A]
Hi,

 When trying to run the example "MyApp.Skia.Linux.FrameBuffer" on my embedded linux device, it errors out on "Failed to set 32 bits display mode"

Specs:
- 32 bits ARM
- Ubuntu 20.04
- .NET 6 (framework dependent)

**Comment by jeromelaban** [comment]
Thanks for the report. Here's the location of the exception:
https://github.com/unoplatform/uno/blob/3fe3862b270b99dbec4d830b547942af61b1a1d9/src/Uno.UI.Runtime.Skia.Linux.FrameBuffer/FrameBufferDevice.cs#L86

This would mean that your device is somehow not allowing a 32bits mode for the framebuffer. Which device are you trying to run on?

**Reply by jeromelaban** [reply]
> I can't really share more hardware specs due to confidentiality.
> I'm pretty sure the display has 16 bits per pixel. Any way to get it to run on that?

No worries. At this time there isn't, as the rest of the rendering assumes a 32 bits buffer, as 16bits color devices are not common, and we work based on common requests or professional support.

For now, we're assuming this pixel format: https://github.com/unoplatform/uno/blob/3fe3862b270b99dbec4d830b547942af61b1a1d9/src/Uno.UI.Runtime.Skia.Linux.FrameBuffer/FrameBufferDevice.cs#L42

But it should certainly be changed to use 16bits with `SKColorType.Rgb565`, for instance. You can try building Uno and changing the framebuffer initialization parameters. For building and debugging Uno, see https://platform.uno/docs/articles/uno-development/debugging-uno-ui.html.

**Comment by TheDutchDev** [comment]
I can't really share more hardware specs due to confidentiality.

I'm pretty sure the display has 16 bits per pixel. Any way to get it to run on that?

**Comment by TheDutchDev** [comment]
Hi,

I have some more info I can share now since I tried it on a more widely used platform which resulted in the same issue.

I tried running the same example project on a raspberry pi 4 (8GB) with the official raspberry pi 7 inch display on linux framebuffer mode (raspbian lite). 

RPI:
https://www.raspberrypi.com/products/raspberry-pi-4-model-b/specifications/

Display:
https://www.raspberrypi.com/products/raspberry-pi-touch-display/

- 64 bit ARM raspbian lite
- .NET 6

These are pretty widely used for raspberry pi projects. Any reason why it would still throw the same error here as well/why uno doesn't seem to support it?

**Reply by jeromelaban** [reply]
As I mentioned in https://github.com/unoplatform/uno/discussions/8737#discussioncomment-2723477, there's no specific reason for this mode not to be supported, just the assumption that most devices would support 32bits. 

You can try making the modification to the Uno code and see how this can help.

**Reply by jeromelaban** [reply]
This will be fixed by https://github.com/unoplatform/uno/pull/9418
